src/utils/intersection-observer-3p-host.js

```diff
@@ -71,21 +72,37 @@ const INIT_TIME = Date.now();
 
 /**
  * A function to get the element's current IntersectionObserverEntry
- * regardless of the intersetion ratio. Only available when element is not
+ * regardless of the intersection ratio. Only available when element is not
  * nested in a container iframe.
  * @param {!../layout-rect.LayoutRectDef} element element's rect
  * @param {?../layout-rect.LayoutRectDef} owner element's owner rect
  * @param {!../layout-rect.LayoutRectDef} hostViewport hostViewport's rect
  * @return {!IntersectionObserverEntry} A change entry.
  */
-export function getIntersectionChangeEntry(element, owner, hostViewport) {
+function getIntersectionChangeEntryHelper(element, owner, hostViewport) {
   const intersection =
     rectIntersection(element, owner, hostViewport) ||
     layoutRectLtwh(0, 0, 0, 0);
   const ratio = intersectionRatio(intersection, element);
   return calculateChangeEntry(element, hostViewport, intersection, ratio);
 }
 
+/**
+ * A function to get the element's current IntersectionObserverEntry.
+ * @param {AmpElement} element
+ * @return {!IntersectionChangeEntry} a change entry
+ */
+export function getIntersectionChangeEntry(element) {
+  const box = element.getLayoutBox();
```

Comment:
This is intended to get the bounding client rect of the ad iframe, not the amp-element. Are we sure the size is the same?

Also, this is an absolutely positioned (from the top-left corner of the `<body>`) box, and not a viewport-relative box, which is extremely important for ads (they want to know that they're in viewport, not 10m pixels down the page). We can accommodate that in `getIntersectionChangeEntryHelper` by subtracting the `viewportBox`'s position.